src/io/command_manager.rs

```diff
@@ -21,7 +53,8 @@ pub struct CommandManager {
     rcon_password: Arc<str>,
     rcon: Option<Connection<TcpStream>>,
     rcon_port: u16,
-
+    error_state: ErrorState,
+    error_hist: ErrorState,
```

Comment:
Minor style nitpick:
Their names could be slightly more descriptive, I think something like `current_error` and `last_error` would make more sense, but doesn't have to be that.

Reply:
see 8dbd8ed